src/spec/gocli/integration/cross_release_collocation_spec.rb

```diff
@@ -22,6 +22,69 @@
         }]
       }
     end
+    let(:manifest_for_properties) do
+      {
+        'releases' => [
+          {'name' => 'dummy', 'version' => 'latest'},
+          {'name' => 'bosh-release', 'version' => '0.1-dev'},
+        ],
+        'jobs' => [{
+          'name' => 'foobar',
+          'templates' => [
+            {'name' => 'dummy_with_properties', 'release' => 'dummy'},
+            {'name' => 'foobar', 'release' => 'bosh-release'},
+          ],
+          'resource_pool' => 'a',
+          'instances' => 1,
+          'networks' => [{'name' => 'a'}]
+        }]
+      }
+    end
+
+    it 'shows correct versions and names in templates' do
+      bosh_runner.run("upload-release #{spec_asset('dummy-release.tgz')}")
+      bosh_runner.run("upload-release #{spec_asset('bosh-release-0+dev.1.tgz')}")
+      bosh_runner.run("upload-stemcell #{spec_asset('valid_stemcell.tgz')}")
+
+      cloud_config_manifest = yaml_file('cloud_manifest', Bosh::Spec::Deployments.simple_cloud_config)
+      bosh_runner.run("update-cloud-config #{cloud_config_manifest.path}")
+      manifest_hash = Bosh::Spec::Deployments.simple_manifest.merge(manifest_for_properties)
+
+      # We manually change the deployment manifest release version, because of weird issue where
+      # the uploaded release version is `0+dev.1` and the release version in the deployment manifest
+      # is `0.1-dev`, similar to links_spec
+      manifest_hash['releases'][1]['version'] = '0+dev.1'
```

Comment:
seems weird and distracting and it should be fixed sometime, but outside of scope of this change; thanks for the reminder